flink-core/src/main/java/org/apache/flink/api/common/functions/RuntimeContext.java

```diff
@@ -473,4 +520,26 @@ <IN, ACC, OUT> AggregatingState<IN, OUT> getAggregatingState(
      */
     @PublicEvolving
     <UK, UV> MapState<UK, UV> getMapState(MapStateDescriptor<UK, UV> stateProperties);
+
+    /**
+     * Get the meta information of current job.
+     *
+     * @return the job meta information.
+     */
+    @PublicEvolving
+    @Nullable
+    default JobInfo getJobInfo() {
+        return null;
+    }
+
+    /**
+     * Get the meta information of current task.
+     *
+     * @return the task meta information.
+     */
+    @PublicEvolving
+    @Nullable
+    default TaskInfo getTaskInfo() {
+        return null;
+    }
```

Comment:
Why provide default implementation for these methods? And why these are `@Nullable`? What does it mean when jobinfo / taskinfo is null?

Reply:
I've changed the default methods `getJobInfo()` and `getTaskInfo()` to be abstract.